metrics/api/src/main/java/io/helidon/metrics/api/NoOpMeterRegistry.java

```diff
@@ -108,21 +113,38 @@ public Iterable<String> scopes() {
     @Override
     public <B extends Meter.Builder<B, M>, M extends Meter> M getOrCreate(B builder) {
         NoOpMeter.Builder<?, ?> b = (NoOpMeter.Builder<?, ?>) builder;
-        return findOrRegister(NoOpMeter.Id.create(b.name(),
-                                                  b.tags()),
-                              builder);
+        M result = findOrRegister(NoOpMeter.Id.create(b.name(),
+                                                      b.tags()),
+                                  builder);
+        onAddListeners.forEach(listener -> listener.accept(result));
+        return result;
     }
 
     @Override
     public MeterRegistry onMeterAdded(Consumer<Meter> listener) {
+        onAddListeners.add(listener);
```

Comment:
Shouldn't this be guarded by a lock? I guess this can be multi-threaded. And if you call this and `onMeterRemoved` concurrently, you would get a concurrent modification exception.

Reply:
Changed.

Both lists of listeners (onAdd and onRemove) are now `CopyOnWriteArrayList`, here and in `MMeterRegistry`. 

Update operations will be very rare compared to iterating over the lists.